### PR TITLE
C++ Solution using Prim's for Portals

### PR DESCRIPTION
**Pull request by maggie-j-liu**
Fixed stuff i messed up in the previous pr, added a solution using Prim's.

The general idea is the same as the official analysis, just implemented with Prim's instead of Kruskal's, so I didn't think I needed to add an explanation. I can write one if it's needed, though.

### CODE REVIEW

File: solutions/usaco-1138.mdx [hunk 1]
```diff
@@ -101,4 +103,77 @@ int main()
...
+		adj[p2].push_back({p1, 0});
+		adj[p3].push_back({p4, 0});
+		adj[p4].push_back({p3, 0});
+		// to get an edge from p1 to p3, we need to pay to permute the portals
+		adj[p1].push_back({p3, cost});
+		adj[p3].push_back({p1, cost});
+	}
+	pq.push({0, 0});
+	bool visited[2 * MAX_N]{};
+	int ans = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-1138.mdx:153]: i'd put min_cost instead of ans
but you do you ig

File: solutions/usaco-1138.mdx [hunk 2]
```diff
@@ -101,4 +103,77 @@ int main()
...
+	while (!pq.empty())
```
Comment (SansPapyrus683) [inline solutions/usaco-1138.mdx:154]: maybe a simple comment
`// run Prim's on the constructed graph` or smth

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
out of all the other content authors your code is the best tbh

File: solutions/usaco-1138.mdx (hunk 1 above)
Comment (maggie-j-liu) [reply solutions/usaco-1138.mdx:153]: i think `ans` is fine...

Comment (SansPapyrus683) [reply solutions/usaco-1138.mdx:153]: more explicit, and i don't think it would drastically affect conciseness or readability

Comment (maggie-j-liu) [reply solutions/usaco-1138.mdx:153]: ok

**Review by dongliu0426 — APPROVED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**